### PR TITLE
Kernel/ACPI: Simplify parser initialization

### PR DESCRIPTION
**Pull request by supercomputer7**
Let's remove the DynamicParser class, as it really did nothing yet in
the Kernel. Instead, when we add support for AML parsing, we can figure
out how to do it properly without the need of a derived class that just
complicates everything for no good reason.

### CODE REVIEW

**Comment by supercomputer7** [comment]
The PR name and commit title were too bombastic, anyway, this commit still greatly simplifies the subsystem.

File: Kernel/ACPI/Initialize.cpp [hunk 1]
```diff
@@ -21,10 +22,15 @@ UNMAP_AFTER_INIT void initialize()
...
-    else
-        Parser::initialize<Parser>(rsdp.value());
+    auto facp = StaticParsing::find_table(rsdp.value(), "FACP");
+    if (!facp.has_value())
+        return;
+    auto facp_table = Memory::map_typed<Structures::FADT>(facp.value());
+    u8 irq_line = facp_table->sci_int;
+
+    Parser::initialize(rsdp.value(), facp.value(), irq_line);
+    VERIFY(Parser::the());
```
Comment (IdanHo) [inline Kernel/ACPI/Initialize.cpp:32]: If `initialize` returns void one would assume that it cant fail, but we then VERIFY that it didn't in a round-about way, we should either make `initialize` VERIFY that it didnt initially, or make it return a check-able error code/result etc.

File: Kernel/ACPI/Initialize.cpp [hunk 2]
```diff
@@ -21,10 +22,15 @@ UNMAP_AFTER_INIT void initialize()
...
+    Parser::the()->set_feature_level(kernel_command_line().acpi_feature_level());
```
Comment (IdanHo) [inline Kernel/ACPI/Initialize.cpp:33]: As far as I'm aware the acpi feature level is constant for the lifetime of the Kernel, so there's no reason to have a setter for it, can we just pass it in the constructor instead/

**Review by IdanHo — COMMENTED** [review]
This change indeed makes it nicer, a couple of comments though:

**Review by IdanHo — APPROVED** [review]
LGTM (You might want to bump the copyright year in `Initialize.cpp`, `Parser.cpp` and `Parser.h`)

**Pull request merged**